examples/stm32/xively_demo_stm32f4xx_nucleo_wifi/Src/main.c

```diff
@@ -90,90 +100,51 @@ char* ssid          = "SSID";
 char* seckey        = "PASSWORD";
 WiFi_Priv_Mode mode = WPA_Personal;
 
+
 #define XI_ACCOUNT_ID "XIVELY ACCOUNT ID"
 #define XI_DEVICE_ID "XIVELY DEVICE ID"
 #define XI_DEVICE_PASS "XIVELY DEVICE PASSWORD"
 
 /* the interval for the time function */
 #define XI_PUBLISH_INTERVAL_SEC 5
 
-typedef struct topic_descr_s
-{
-    const char* const name;
-} mqtt_topic_descr_t;
-
 /* declaration of fn pointer for the msg handler */
 typedef void ( *on_msg_handler_fn_t )( const uint8_t* const msg, size_t msg_size );
 
-/* declaration of fn pointer for the polling push mechanism */
-typedef xi_state_t ( *on_msg_push_fn_t )(
-    const mqtt_topic_descr_t* const mqtt_topic_descr );
+/* declaration of fn pointer for the publishing functions */
+typedef xi_state_t ( *on_msg_push_fn_t )( const char* mqtt_topic_descr );
 
 /* each topic will have it's descriptor and the handler in case of the message */
 typedef struct mqtt_topic_configuration_s
 {
-    mqtt_topic_descr_t topic_descr;
+    const char* const topic_name;
     on_msg_handler_fn_t on_msg_pull;
     on_msg_push_fn_t on_msg_push;
 } mqtt_topic_configuration_t;
 
 /* combines name with the account and device id */
-#define XI_TOPIC_NAME_MANGLE( name ) \
+#define XI_TOPIC_NAME_MANGLE( name )                                                     \
     "xi/blue/v1/" XI_ACCOUNT_ID "/d/" XI_DEVICE_ID "/" name
 
 /* declaration of topic handlers for SUB'ed topics */
 static void on_led_msg( const uint8_t* const msg, size_t msg_size );
 
 /* declaration of topic handlers for PUB'ed topics */
-static xi_state_t pub_accelerometer( const mqtt_topic_descr_t* const mqtt_topic_descr );
-static xi_state_t pub_magnetometer( const mqtt_topic_descr_t* const mqtt_topic_descr );
-static xi_state_t pub_barometer( const mqtt_topic_descr_t* const mqtt_topic_descr );
-static xi_state_t pub_button( const mqtt_topic_descr_t* const mqtt_topic_descr );
-static xi_state_t pub_gyroscope( const mqtt_topic_descr_t* const mqtt_topic_descr );
-static xi_state_t pub_humidity( const mqtt_topic_descr_t* const mqtt_topic_descr );
-static xi_state_t pub_temperature( const mqtt_topic_descr_t* const mqtt_topic_descr );
+static xi_state_t pub_accelerometer( const char* topic_name );
+static xi_state_t pub_magnetometer( const char* topic_name );
+static xi_state_t pub_barometer( const char* topic_name );
+static xi_state_t pub_button( const char* topic_name );
+static xi_state_t pub_gyroscope( const char* topic_name );
+static xi_state_t pub_humidity( const char* topic_name );
+static xi_state_t pub_temperature( const char* topic_name );
 
 /* topic's initialisation function */
 static xi_state_t init_xively_topics( xi_context_handle_t in_context_handle );
 
-/* table of topics used for this demo */
-static const mqtt_topic_configuration_t topics_array[] = {
-    {{XI_TOPIC_NAME_MANGLE( "Accelerometer" )}, NULL, pub_accelerometer},

```

Comment:
Because it's way too complicated for a static demo.  The users has to fight their way through an abstracted table system for their "ease of use" but I don't think it's actually easy.  The table includes confusing fields including in going and outgoing data.  Isn't it easier to use pub when you want to pub in the logic and subscribe to the one topic after the connection then to conflate the two in one abstract table ? I think so.